src/core/qgsapplication.cpp

```diff
@@ -1532,6 +1532,19 @@ bool QgsApplication::createDatabase( QString *errorMessage )
       }
       return false;
     }
+
+    QFile::Permissions perms = QFile( qgisPrivateDbFile.fileName() ).permissions();
+    if ( !( perms & QFile::WriteOwner ) )
+    {
+      if ( !qgisPrivateDbFile.setPermissions( perms | QFile::WriteOwner ) )
+      {
+        if ( errorMessage )
+        {
+          *errorMessage = tr( "[ERROR] Can not make qgis.db user writable" );
```

Comment:
I would add the path here and remove the `[ERROR]` prefix.

Reply:
Done, I have updated the error message.